src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -187,6 +186,45 @@ public Object createModule(String name) throws CheckstyleException {
     }
 
     /**
+     * Determines whether the class can be loaded successfully.
+     * @param name name of module.
+     * @return true if the name is in the set or the class has been loaded.
+     * @throws CheckstyleException if  fails to call isLoaded.
+     */
+    private boolean successToLoad(String name) throws CheckstyleException {
+        boolean returnValue = false;
+        //if the name is in the set or has been loaded, return true
+        if (NAME_TO_FULL_MODULE_NAME.containsValue(name) || isLoaded(name)) {
+            returnValue = true;
+        }
+        if (thirdPartyNameToFullModuleNames != null
+            && thirdPartyNameToFullModuleNames.containsValue(Collections.singleton(name))) {
+            returnValue = true;
+        }
+        return returnValue;
+    }
+
+    /**
+     * Determines whether the class has been loaded.
+     * @param name name of module.
+     * @return true if class has been loaded.
+     * @throws CheckstyleException if  fails to call java.lang.findLoadedClass.
+     */
+    private boolean isLoaded(String name) throws CheckstyleException {
```

Comment:
@romani I use function `isLoaded` for checking whether the class has been loaded.Because some test classes have been loaded and their name are not in the collection.For example,[code](https://github.com/checkstyle/checkstyle/blob/546bae162b3e3d59cc53ff6e1f3cc0d8983e3692/src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java#L259).And I think it is not a good solution with two points:
1.Using reflection and hard to pass Coverage check.
2.It is redundant with `Class.forName`.
Now,my idea is:
Refactoring test case,all test cases should create `Object` of the `Test Class` with a standard form and the [case](https://github.com/checkstyle/checkstyle/blob/546bae162b3e3d59cc53ff6e1f3cc0d8983e3692/src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java#L259) is forbidden. 
Please give me some suggestion.

Reply:
＠romani I'm free now and I can finish this PR.

Reply:
@djydewang ,

> Because some test classes have been loaded and their name are not in the collection

not is collection, it mean, that it should not be loaded, instantiation should fail, like it is defined in test.

> I use function isLoaded

I did not expect such code to appear, please explain why it is required in a bit more details.

Reply:
@romani
More examples,
[case1](https://github.com/checkstyle/checkstyle/blob/08793f2816a2ce275afdb89f2a39fa535e0cf3e6/src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java#L813) [case2](https://github.com/checkstyle/checkstyle/blob/08793f2816a2ce275afdb89f2a39fa535e0cf3e6/src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java#L607)
I mean these classes have been loaded to JVM when we want to create objects of them.And they are not in the collection.Because they are not in main code or treated as thirdPartyClasses.So I use isLoaded to predict cases whether have success to load class by reflection at createObject. And the [case](https://github.com/checkstyle/checkstyle/blob/546bae162b3e3d59cc53ff6e1f3cc0d8983e3692/src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java#L259) should be loaded.First the load and the instantiation.

Reply:
Loading of Checks that are not in main sources, should be loaded by full name, if createModuleConfig  do this in different way that should be changed.
Such cases should be instantiated by createObject .

Reply:
So,createObject should be public?I will remove isLoaded and refactor test cases.